presto-main/src/main/java/io/prestosql/spiller/FileSingleStreamSpillerFactory.java

```diff
@@ -183,4 +189,21 @@ private boolean hasEnoughDiskSpace(Path path)
             throw new PrestoException(OUT_OF_SPILL_SPACE, "Cannot determine free space for spill", e);
         }
     }
+
+    private boolean isAccessible(Path path)
+    {
+        return isReadable(path) && isWritable(path) && isExecutable(path);
```

Comment:
'x' bit is needed to create files and sub-directories inside the given directory. Also note that, prior to this change the constructor of `FileSingleStreamSpillerFactory` was only checking for 'w' bit, which is a necessary but not sufficient condition. So that bug has to be fixed regardless of this PR.

Further reading https://unix.stackexchange.com/questions/149184/folder-with-only-write-permission-is-useless-right